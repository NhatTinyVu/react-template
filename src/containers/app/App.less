@import '~antd/lib/style/themes/default.less';

.layout {
  position: relative;
  min-height: 100vh;
  background-color: #eaeff2;
}

.logo {
  height: 64px;
  float: left;

  img {
    height: 100%;
  }
}

.navbar {
  line-height: 64px;
}

.content {
  display: flex;
  height: 100%;
}

.page {
  flex-grow: 1;
  background-color: white;
}
