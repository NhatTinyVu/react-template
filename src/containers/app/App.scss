.layout {
  position: relative;
  min-height: 100vh !important;
}

.navbar {
  line-height: 64px;
}