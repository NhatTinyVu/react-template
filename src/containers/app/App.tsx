import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Layout } from 'antd';
import Logo from './logo';
import NavBar from './navbar';
import Sider from './sider';
import Pages from './pages';
import Footer from './footer';
import styles from './App.less';

const { Header, Content } = Layout;

const App: React.FC = () => (
  <Router>
    <Layout className={styles.layout}>
      <Header>
        <Logo />
        <NavBar />
      </Header>
      <Content className={styles.content}>
        <Layout className={styles.page}>
          <Sider />
          <Layout>
            <Pages />
            <Footer />
          </Layout>
        </Layout>
      </Content>
    </Layout>
  </Router>
);

export default App;
