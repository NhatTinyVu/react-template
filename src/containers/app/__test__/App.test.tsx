import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from '..';

describe('App', () => {
  test('renders correctly', (): void => {
    const wrapper = shallow(<App />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
