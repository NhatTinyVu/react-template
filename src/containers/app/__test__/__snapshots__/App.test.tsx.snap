// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<BrowserRouter>
  <Adapter
    className="layout"
  >
    <Adapter>
      <Logo />
      <NavBar />
    </Adapter>
    <Adapter
      className="content"
    >
      <Adapter
        className="page"
      >
        <Sider />
        <Adapter>
          <NavBar />
          <Footer />
        </Adapter>
      </Adapter>
    </Adapter>
  </Adapter>
</BrowserRouter>
`;
